lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3FileSystemLoader.java

```diff
@@ -167,6 +177,60 @@ else if (iamRole.isPresent()) {
         };
     }
 
+    private static S3PreSignerFactory s3PreSignerFactory(SdkHttpClient httpClient, OpenTelemetry openTelemetry, S3FileSystemConfig config)
+    {
+        Optional<AwsCredentialsProvider> staticCredentialsProvider = createStaticCredentialsProvider(config);
+        Optional<String> staticRegion = Optional.ofNullable(config.getRegion());
+        Optional<String> staticEndpoint = Optional.ofNullable(config.getEndpoint());
+        boolean pathStyleAccess = config.isPathStyleAccess();
+        boolean useWebIdentityTokenCredentialsProvider = config.isUseWebIdentityTokenCredentialsProvider();
+        Optional<String> staticIamRole = Optional.ofNullable(config.getIamRole());
+        String staticRoleSessionName = config.getRoleSessionName();
+        String externalId = config.getExternalId();
+
+        return mapping -> {
```

Comment:
We don't need to support S3 security mapping for this, so we can simplify this code.

Reply:
Ok